File: app/src/main/java/com/kota/hh/injection/component/ConfigPersistentComponent.java
package com.kota.hh.injection.component;

import com.kota.hh.injection.ConfigPersistent;
import com.kota.hh.injection.module.ActivityModule;

import dagger.Component;

@ConfigPersistent
@Component(dependencies = ApplicationComponent.class)
public interface ConfigPersistentComponent {

    ActivityComponent activityComponent(ActivityModule activityModule);

}

File: app/src/main/java/com/kota/hh/injection/component/ConfigPersistentComponent.kt
package com.kota.hh.injection.component

import com.kota.hh.injection.ConfigPersistent
import com.kota.hh.injection.module.ActivityModule
import dagger.Component

@ConfigPersistent
@Component(dependencies = arrayOf(ApplicationComponent::class))
interface ConfigPersistentComponent {

    fun activityComponent(activityModule: ActivityModule): ActivityComponent

}